org.civicrm.reqangular/test/models/contract_test.js

```diff
@@ -0,0 +1,79 @@
+/* eslint-env amd, jasmine */
+
+define([
+  'common/lodash',
+  'common/angularMocks',
+  'common/models/contract',
+  'common/mocks/services/hr-settings-mock',
+  'common/mocks/services/api/contract-mock',
+  'common/mocks/models/instances/contract-instance-mock'
+], function (_) {
+  'use strict';
+
+  describe('Contract', function () {
+    var $provide;
+    var $rootScope;
+    var Contract;
+    var ContractInstanceMock;
+    var contractAPI;
+    var contractAPIMock;
+    var contracts;
+
+    beforeEach(function () {
+      module('common.models', 'common.mocks', function (_$provide_) {
+        $provide = _$provide_;
+      });
+      inject([
+        'api.contract.mock', 'HR_settingsMock',
+        function (_contractAPIMock_, HRSettingsMock) {
+          contractAPIMock = _contractAPIMock_;
+          $provide.value('api.contract', contractAPIMock);
+          $provide.value('HR_settings', HRSettingsMock);
+        }
+      ]);
+    });
+
+    beforeEach(inject([
+      '$rootScope', 'Contract', 'ContractInstanceMock',
+      'api.contract',
+      function (_$rootScope_, _Contract_, _ContractInstanceMock_, _contractAPI_) {
+        $rootScope = _$rootScope_;
+
+        Contract = _Contract_;
+
+        ContractInstanceMock = _ContractInstanceMock_;
+
+        contractAPI = _contractAPI_;
+
+        contractAPI.spyOnMethods();
+
+        contracts = contractAPI.all();
+      }
+    ]));
+
+    it('has the expected api', function () {
+      expect(Object.keys(Contract)).toEqual(['all']);
+    });
+
+    describe('all()', function () {
+      describe('instances', function () {
+        it('returns a list of model instances', function (done) {
+          Contract.all().then(function (response) {
```

Comment:
re: does this mean this file is coded in a wrong way?
Its not wrong , but this file was coded long back and after that the conventions we follow has changed a lot as we found more expressive and efficient way to write unit test cases. Those rules are followed in L&A(the example I provided)

1. Any kind of setup required for the unit test should be moved to `beforeEach`
2. Any kind steps required after the unit test finishes should be moved to `afterEach`
3. The it block should contain only the expect statements, or the lines which are absolutely required to run the test. That is what happened in the example you have provided, without the then statement, its impossible(I think?) to write the unit test. So it was kept.
4. Every expect should contain a single expect statement.